### PR TITLE
Fix/rename altitude difference to elevation gain

### PR DESCRIPTION
**Pull request by davidholzwarth**
# Summary

Changed `Altitude difference` in the Map screen to `Elevation gain`, and updated the calculations accordingly.

# Details

A calculation for elevation gain is already made in the DetailedHikeRoute class, so to remove redundancy, I moved the DetailedHikeRoute functions to a separate opject, `RouteUtils`, which now contains five functions which calculate useful information about hikes (elevation gain, estimated time, etc.). Currently, the DetailedHikeRoute uses these Utils for getting the "details" about its hike, and the MapScreen. 

The tests were changed accordingly.

### CODE REVIEW

File: app/src/main/java/ch/hikemate/app/ui/components/HikeCard.kt [hunk 1]
```diff
@@ -115,12 +115,11 @@ fun HikeCard(
                           style = MaterialTheme.typography.bodySmall)
                       Text(
                           text =
-                              if (altitudeDifference == null)
+                              if (elevationGain == 0.0)
                                   stringResource(R.string.hike_card_no_data_label)
```
Comment (rBergCode) [inline app/src/main/java/ch/hikemate/app/ui/components/HikeCard.kt:119]: Warning: If a hike has a real `elevationGain` of 0m we don't want to show "no_data". I think we should keep the convention of `elevationGain` to be null if we don't have any data.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -6,8 +6,8 @@
     <string name="default_web_client_id">154591189513-eilbr2jtms0av2ses4tsvo62009q5t7n.apps.googleusercontent.com</string>
 
     <!-- Hike Card -->
-    <string name="hike_card_altitude_label">Altitude difference</string>
-    <string name="hike_card_altitude_value_template">%sm</string>
+    <string name="hike_card_altitude_label">Elevation gain</string>
+    <string name="hike_card_altitude_value_template">%.0fm</string>
```
Comment (rBergCode) [inline app/src/main/res/values/strings.xml:10]: Nitpick: You should change the name of the strings to reflect the change.

File: app/src/main/java/ch/hikemate/app/utils/RouteUtils.kt
Comment (rBergCode) [inline app/src/main/java/ch/hikemate/app/utils/RouteUtils.kt:1]: Praise: Nice! This is way cleaner and easier to use.

File: app/src/test/java/ch/hikemate/app/utils/RouteUtilsTest.kt [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+    assertEquals(207.59, RouteUtils.computeTotalDistance(points), 0.1)
+  }
+
+  @Test
+  fun test_computeTotalDistance_returns_zero_for_empty_list() {
+    val points = emptyList<LatLong>()
+    val totalDistance = RouteUtils.computeTotalDistance(points)
+    assertEquals(0.0, totalDistance, 0.1)
+  }
```
Comment (rBergCode) [inline app/src/test/java/ch/hikemate/app/utils/RouteUtilsTest.kt:85]: Praise: Nice edge case test!

File: app/src/test/java/ch/hikemate/app/utils/RouteUtilsTest.kt
Comment (rBergCode) [inline app/src/test/java/ch/hikemate/app/utils/RouteUtilsTest.kt:1]: Praise: Really nice unit test. Good job!

**Review by rBergCode — CHANGES_REQUESTED** [review]
Good job! Really clean PR. 
I have some suggestions for you to address before merging.
